pkg/datasource/data.go

```diff
@@ -159,28 +160,43 @@ type dataSource struct {
 
 	byteOrder binary.ByteOrder
 	lock      sync.RWMutex
+
+	config *viper.Viper
 }
 
-func newDataSource(t Type, name string) (*dataSource, error) {
+func newDataSource(t Type, name string, options ...DataSourceOption) (*dataSource, error) {
 	switch t {
 	case TypeSingle, TypeArray:
 	default:
 		return nil, fmt.Errorf("invalid data source type: %v", t)
 	}
 
-	return &dataSource{
+	ds := &dataSource{
 		name:            name,
 		dType:           t,
 		requestedFields: make(map[string]bool),
 		fieldMap:        make(map[string]*field),
 		byteOrder:       binary.NativeEndian,
 		tags:            make([]string, 0),
 		annotations:     map[string]string{},
-	}, nil
+	}
+
+	for _, option := range options {
+		option(ds)
+	}
+
+	if ds.config != nil {
+		// Apply configuration
+		annotations := ds.config.GetStringMapString("annotations")
+		for k, v := range annotations {
+			ds.annotations[k] = v
+		}
+	}
+	return ds, nil
 }
 
-func New(t Type, name string) (DataSource, error) {
-	return newDataSource(t, name)
+func New(t Type, name string, options ...DataSourceOption) (DataSource, error) {
+	return newDataSource(t, name, options...)
 }
 
 func NewFromAPI(in *api.DataSource) (DataSource, error) {
```

Comment:
Load in `NewFromAPI` also the annotations (also the tags), otherwise they will be sent from server to client, but operators in the client side will not see them.

Reply:
Isn't it already done? The `(*field)(f)` refers to field that contains annotations and tags.

Reply:
You're right. These are annotations on the fields, not in the ds itself.